### PR TITLE
add mongodb backup script

### PR DESCRIPTION
**Pull request by fredsmith**

### CODE REVIEW

File: playbooks/roles/mongo/templates/backup-mongo-to-s3.j2 [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      ;;
+  esac
+done
+
+if [[ "{{ MONGO_S3_BACKUP }}" != "true" ]]; then
+  # only run if explicitly enabled
+  exit
+fi
+
+MYNODENAME=$(echo "db.isMaster()" | mongo -u "{{ COMMON_MONGO_READ_ONLY_USER }}" -p"{{ COMMON_MONGO_READ_ONLY_PASS }}" "{{ EDXAPP_MONGO_DB_NAME }}" | grep \"me\" | cut -f 2 -d ':' | sed -e 's/ //' -e 's/,//');
```
Comment (fredsmith) [inline playbooks/roles/mongo/templates/backup-mongo-to-s3.j2]: db.isMaster() returns something like this:

> connecting to: stage
> {
>   "setName" : "rs0",
>   "setVersion" : 1,
>   "ismaster" : false,
>   "secondary" : true,
>   "hosts" : [
>       "stage-mongo-0.mckinsey.edx.org:27017",
>       "stage-mongo-2.mckinsey.edx.org:27017",
>       "stage-mongo-1.mckinsey.edx.org:27017"
>   ],
>   "primary" : "stage-mongo-1.mckinsey.edx.org:27017",
>   "me" : "stage-mongo-0.mckinsey.edx.org:27017",
>   "maxBsonObjectSize" : 16777216,
>   "maxMessageSizeBytes" : 48000000,
>   "maxWriteBatchSize" : 1000,
>   "localTime" : ISODate("2014-08-15T16:02:23.686Z"),
>   "maxWireVersion" : 2,
>   "minWireVersion" : 0,
>   "ok" : 1
> }

File: playbooks/roles/mongo/templates/backup-mongo-to-s3.j2 [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ISSECONDARY=$(echo "db.isMaster()" | mongo -u "{{ COMMON_MONGO_READ_ONLY_USER }}" -p"{{ COMMON_MONGO_READ_ONLY_PASS }}" "{{ EDXAPP_MONGO_DB_NAME }}" | grep secondary | cut -f 2 -d ':' | sed -e 's/ //' -e 's/,//')
+if [[ $ISSECONDARY != "true" ]]; then 
+  # backups should be run on secondary server
+  exit; 
+fi
+
+MONGOOUTDIR=$(mktemp -d -p {{ MONGO_S3_BACKUP_TEMPDIR }})
+DATESTAMP=$(date +'%Y-%m-%d-%H%M')
+$noop mongodump --host {{ EDXAPP_MONGO_HOSTS[0] }} -u "{{ COMMON_MONGO_READ_ONLY_USER }}" -p"{{ COMMON_MONGO_READ_ONLY_PASS }}" -o $MONGOOUTDIR
+$noop tar zcf - $MONGOOUTDIR | {{ aws_cmd }} put {{ item['path'] }} "s3://{{ MONGO_S3_BACKUP_BUCKET }}/{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}/mongo-backup-$DATESTAMP.tar.gz"
```
Comment (fredsmith) [inline playbooks/roles/mongo/templates/backup-mongo-to-s3.j2]: that item['path'] shouldn't be there...

File: playbooks/roles/mongo/tasks/main.yml [hunk 3]
```diff
@@ -76,3 +76,39 @@
     state=present
   with_items: MONGO_USERS
   when: mongo_create_users
+
```
Comment (e0d) [inline playbooks/roles/mongo/tasks/main.yml:79]: Was there a reason to not use awscli which is installed already?

File: playbooks/roles/mongo/tasks/main.yml [hunk 4]
```diff
@@ -76,3 +76,39 @@
...
+- name: install backup-mongo-to-s3 script
+  template: >
+    src=backup-mongo-to-s3.j2
+    dest=/edx/bin/backup-mongo-to-s3.sh
+    owner=root
+    group=root
+    mode=0700
+  when: MONGO_S3_BACKUP
+
+- name: schedule backup-mongo-to-3s crontab
```
Comment (e0d) [inline playbooks/roles/mongo/tasks/main.yml:105]: Are we using dean man's snitch or something?

File: playbooks/roles/mongo/templates/backup-mongo-to-s3.j2 [hunk 5]
```diff
@@ -0,0 +1,68 @@
+{% set lb = '{' %}
+{% set rb = '}' %}
+#!/bin/bash
```
Comment (e0d) [inline playbooks/roles/mongo/templates/backup-mongo-to-s3.j2:3]: You promised no bash ;)

**Pull request merged**